src/compositions/Timeline/Timeline.tsx

```diff
@@ -1,42 +1,220 @@
+    onSetCurrentTime,
     className = '',
-}): ReactElement => (
-    <div
-        className={`timeline ${className}`}
-        style={{ '--current-time': currentTime } as CSSProperties}
-    >
-
-        <div className="timeline__timings"></div>
-
-        <ul className="timeline__cue-list">
-            {cues.map((cue, index) => (
-                <li key={index}
-                    className={`timeline__cue-item ${activeCueIndex === index ? 'timeline__cue-item--active' : ''}`}
-                    style={{ '--cue-start-time': cue.startTime, '--cue-end-time': cue.endTime } as CSSProperties}
+}): ReactElement => {
+    const [scale, setScale] = useState(5)
+    const [scaleWidth, setScaleWidth] = useState(50)
+    const [timelineData, setTimelineData] = useState<TimelineRow[]>([])
+
+    const timelineState = useRef<TimelineState>(null)
+
+    const timelineEffects: Record<string, TimelineEffect> = {
+        caption: {
+            id: 'caption',
+            name: 'caption',
+        },
+        video: {
+            id: 'video',
+            name: 'video',
+        },
+    }
+
+    const increaseZoom = () => {
+        if (scale > 2) {
+            setScale(scale / 2)
+        } else {
+            setScaleWidth(scaleWidth * 2)
+        }
+    }
+
+    const decreaseZoom = () => {
+        if (timelineState.current && videoLength * (scaleWidth / Math.floor(scale)) < timelineState.current.target.clientWidth) {
+            return
+        }
+
+        if (scaleWidth > 50) {
+            setScaleWidth(scaleWidth / 2)
+        } else {
+            setScale(scale * 2)
+        }
+    }
+
+    const scrollToTime = (
+        timelineState: TimelineState,
+        time: number,
+        position: number,
+        scale: number,
+        scaleWidth: number,
+        maxScrollTime: number = Infinity,
+    ): void => {
+        const timelineWidth = timelineState.target.clientWidth
+        const maxScroll = maxScrollTime * (scaleWidth / Math.floor(scale)) - timelineWidth + 50
+        const left = Math.min(time * (scaleWidth / Math.floor(scale)) - (timelineWidth * position), maxScroll)
+        timelineState.setScrollLeft(left)
+    }
+
+    useEffect(() => {
+        if (! timelineState.current) {
+            return
+        }
+
+        // Auto scroll while playing
+        timelineState.current.listener.on('setTimeByTick', ({ time }) => {
+            if (timelineState.current) {
+                timelineState.current.setTime(time) // Necessary for smooth scroll
+
+                scrollToTime(timelineState.current, time, 0.8, scale, scaleWidth, videoLength)
+            }
+        })
+
+        // Scroll to current time while zooming
+        scrollToTime(timelineState.current, timelineState.current.getTime(), 0.5, scale, scaleWidth)
+
+        return () => {
+            if (! timelineState.current) {
+                return
+            }
+
+            timelineState.current.listener.offAll()
+        }
+    }, [scale, scaleWidth])
```

Comment:
Please be aware that the event listener will be set (and removed) every time `scale` and/or `scaleWidth` changes.

As the event handler method relies on these values, I reckon this was probably intended.